modules/aws/aws.go

```diff
@@ -414,6 +431,122 @@ The account %q has been removed from %q.`, uid, r.p.AccountName)),
+			r.p.AccountName, uid, err)
+		return
+	}
+	if !r.Conf.ApplyChanges {
+		log.Printf("[dryrun] aws %q: would have reset AWS IAM user %q with password %q",
+			r.p.AccountName, uid, password)
+		goto notify
+	}
+	glpo, err = r.cli.GetLoginProfile(&iam.GetLoginProfileInput{
+		UserName: aws.String(uid),
+	})
+	if err != nil {
+		log.Printf("[error] aws %q: failed to create login profile for user %q: %v",
+			r.p.AccountName, uid, err)
+		return
+	}
+	if glpo == nil {
+		clpo, err = r.cli.CreateLoginProfile(&iam.CreateLoginProfileInput{
+			Password:              aws.String(password),
+			UserName:              aws.String(uid),
+			PasswordResetRequired: aws.Bool(true),
+		})
+		if err != nil || clpo == nil {
+			log.Printf("[error] aws %q: failed to create login profile for user %q: %v",
+				r.p.AccountName, uid, err)
+			return
+		}
+	} else {
+		ulpo, err = r.cli.UpdateLoginProfile(&iam.UpdateLoginProfileInput{
+			Password:              aws.String(password),
+			UserName:              aws.String(uid),
+			PasswordResetRequired: aws.Bool(true),
+		})
+		if err != nil || ulpo == nil {
+			log.Printf("[error] aws %q: failed to update login profile for user %q: %v",
+				r.p.AccountName, uid, err)
+			return
+		}
+	}
+	if r.p.CreateAccessKey {
+		laao, err = r.cli.ListAccessKeys(&iam.ListAccessKeysInput{
+			UserName: aws.String(uid),
+		})
+		if err != nil || laao == nil {
+			log.Printf("[error] aws %q: failed to list access keys for user %q: %v",
+				r.p.AccountName, uid, err)
+			return
+		}
+		// only create an access key if user has fewer than
+		// hardcoded default # of access keys per user per
+		// http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-limits.html
+		if len(laao.AccessKeyMetadata) > accessKeyLimit {
+			log.Printf("[warning] aws %q: %q already has max of %d access keys",
+				r.p.AccountName, uid, accessKeyLimit)
+		} else {
+			cako, err = r.cli.CreateAccessKey(&iam.CreateAccessKeyInput{
+				UserName: aws.String(uid),
+			})
+			if err != nil || cako == nil {
+				log.Printf("[error] aws %q: failed to create access key for user %q: %v",
+					r.p.AccountName, uid, err)
+				return
+			}
+			accesskey = fmt.Sprintf(`
+	A new access key has been created for you.
+	Add the lines below to ~/.aws/credentials
+	[%s]
+	aws_access_key_id = %s
+	aws_secret_access_key = %s`,
+				r.p.AccountName,
+				*cako.AccessKey.AccessKeyId,
+				*cako.AccessKey.SecretAccessKey)
+		}
+	}
+
+notify:
+	// notify user
+	rcpt := r.Conf.Notify.Recipient
+	if rcpt == "{ldap:mail}" {
+		rcpt = mail
+	}
+
+	body := fmt.Sprintf(`Updated AWS account:
+login: %s
+pass:  %s (change at first login)
+url:   https://%s.signin.aws.amazon.com/console
+%s`, uid, password, r.p.AccountName, accesskey)
+
+	r.Conf.Notify.Channel <- modules.Notification{
+		Module:      "aws",
+		Recipient:   rcpt,
+		Mode:        r.Conf.Notify.Mode,
+		MustEncrypt: true,
+		Body:        []byte(body),
+	}
+	return
+}
```

Comment:
Is there any way you could refactor parts of this and the create user function to avoid code duplication?